### PR TITLE
DBZ-4912 Redis Sink - multi sharded support

### PR DESCRIPTION
**Pull request by spicy-sauce**

### CODE REVIEW

**Review by jpechane — COMMENTED** [review]
@spicy-sauce LGTM thanks. I left one comment.

**Comment by jpechane** [comment]
@gunnarmorling I know it is a bit late but I think we can add this one to the final. LGTM from my PoV.

**Comment by spicy-sauce** [comment]
> @gunnarmorling I know it is a bit late but I think we can add this one to the final. LGTM from my PoV.

It's important to merge this for 1.9 as the current state of Redis Sink has a regression comparing 1.8.
Before using batches, streaming change records one by one worked on multi-sharded Redis database as well.
Once we changed it to Transaction (with Mutli), we gained performance improvement, but the sink no longer works on multi-sharded DB. So, changing to Pipeline (also send the data in a buffer of commands, i.e, batches) is the best way to stream the data in terms of performance and multi-sharded support.
Thanks a lot!

File: debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java [hunk 1]
```diff
@@ -151,46 +158,63 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records,
...
+                        }
+
+                        // Sync the pipeline in Redis and parse the responses (response per command with the same order)
+                        List<Object> responses = pipeline.syncAndReturnAll();
+                        List<ChangeEvent<Object, Object>> processedRecords = new ArrayList<ChangeEvent<Object, Object>>();
+                        int index = 0;
+                        int totalOOMResponses = 0;
+
+                        for (Object response : responses) {
+                            String message = response.toString();
```
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java:183]: >  I'll also open an issue in Debezium JIRA to keep track on this.

Sounds good, thanks.

**Comment by gunnarmorling** [comment]
Makes sense, thanks for providing that context, @spicy-sauce! Squashed and applied.

File: debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (spicy-sauce) [reply debezium-server/debezium-server-redis/src/main/java/io/debezium/server/redis/RedisStreamChangeConsumer.java:183]: Opened [DBZ-4928](https://issues.redhat.com/browse/DBZ-4928)

**Pull request merged**